### PR TITLE
:sparkles: pkg/log/zap: clarify zap level vs. logr verbosity

### PR DESCRIPTION
**Pull request by estroz**
This PR clarifies zap level vs. logr verbosity with a link to the zapr explanation of this difference, and test cases.

/kind documentation

Signed-off-by: Eric Stroczynski <ericstroczynski@gmail.com>

### CODE REVIEW

File: pkg/log/zap/zap.go [hunk 1]
```diff
@@ -101,8 +101,9 @@ func newConsoleEncoder(opts ...EncoderConfigOption) zapcore.Encoder {
 	return zapcore.NewConsoleEncoder(encoderConfig)
 }
 
-// Level sets the the minimum enabled logging level e.g Debug, Info
-// See Options.Level
+// Level sets Options.Level, which configures the the minimum enabled logging level e.g Debug, Info.
+// If using zapcore.Level(N), where N > 0, set zapcore.Level(-1*N) instead.
```
Comment (coderanger) [inline pkg/log/zap/zap.go:105]: This line feels confusing, I think you mean the second bit to be `Level(-1*N)` for how to call this function? Or the first?

**Review by kensipe — APPROVED** [review]
agree with coderanger... this line is hard to parse... otherwise lgtm

File: pkg/log/zap/zap.go (hunk 1 above)
Comment (estroz) [reply pkg/log/zap/zap.go]: Lmk if my update makes more sense.

Comment (coderanger) [reply pkg/log/zap/zap.go]: :+1: Clearer!

**Review by alvaroaleman — APPROVED** [review]
/approve
